expandable-impl/src/macros.rs

```diff
@@ -3,307 +3,314 @@
 
 #[cfg(test)]
 macro_rules! quote {
-    (@inner, ( $( $tt:tt )* ) ) => {
-        $crate::TokenTree::Parenthesed(quote! { $( $tt )* })
+    (@inner $sb:expr, ( $( $tt:tt )* ) ) => {
+        $crate::TokenTree::parenthesed($sb.mk_span(), quote! { @with_sb $sb, $( $tt )* })
     };
 
-    (@inner, { $( $tt:tt )* } ) => {
-        $crate::TokenTree::CurlyBraced(quote! { $( $tt )* })
+
+    (@inner $sb:expr, { $( $tt:tt )* } ) => {
+        $crate::TokenTree::curlyBraced($sb.mk_span(), quote! { @with_sb $sb, $( $tt )* })
     };
 
-    (@inner, @) => {
```

Comment:
Maybe we can group all these `Terminal` creation by calling `quote` another time? This will make next edits much easier.